src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -306,13 +314,67 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                         End If
                     Loop
 
+                    If (compareKind And TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) = 0 Then
+                        Return HasSameTypeArgumentCustomModifiers(DirectCast(t1, NamedTypeSymbol), DirectCast(t2, NamedTypeSymbol))
+                    End If
+
                     Return True
                 End If
             End If
 
             Return (t1 = t2)
         End Function
 
+        Private Function HasSameTypeArgumentCustomModifiers(type1 As NamedTypeSymbol, type2 As NamedTypeSymbol) As Boolean
+            If Not type1.HasTypeArgumentsCustomModifiers() AndAlso Not type2.HasTypeArgumentsCustomModifiers() Then
+                Return True
+            End If
+
+            Dim mod1 = type1.TypeArgumentsCustomModifiers
+            Dim mod2 = type2.TypeArgumentsCustomModifiers
+
+            If mod1.Length <> mod2.Length Then
+                Return False
+            End If
```

Comment:
This feels redundant because if both types have modifiers, the arrays will have the same length because the types are based on the same definition.